app/javascript/packs/leafletMap.js

```diff
@@ -27,263 +27,269 @@ let currentFloor = 0;
-  routeLayer.addTo(map);
-  const uiPadding = getUIPadding();
-  map.fitBounds(polyline.getBounds(), {
-    paddingTopLeft: [uiPadding[0], 0],
-    paddingBottomRight: [0, uiPadding[1]],
-    animate: true,
-    duration: 1,
-  });
+    const route = await $.ajax({
+        type: "GET",
+        url: "/building_map/route",
+        data: `start=${start}&dest=${dest}`,
+        dataType: "json",
+    });
+    if (global.routeLayer) routeLayer.clearLayers();
+    else global.routeLayer = L.layerGroup();
+    const polyline = addPolyline(route["polyline"], routeLayer);
+    addMarkers([route["marker"]], routeLayer);
+    routeLayer.addTo(map);
+    const uiPadding = getUIPadding();
+    map.fitBounds(polyline.getBounds(), {
+        paddingTopLeft: [uiPadding[0], 0],
+        paddingBottomRight: [0, uiPadding[1]],
+        animate: true,
+        duration: 1,
+    });
 }
 
 async function getBuildings() {
-  return $.ajax({
-    type: "GET",
-    url: "/building_map/buildings",
-    dataType: "json",
-  });
+    return $.ajax({
+        type: "GET",
+        url: "/building_map/buildings",
+        dataType: "json",
+    });
 }
 
 async function getBuildingMarkers() {
-  return $.ajax({
-    type: "GET",
-    url: "/building_map/markers",
-    dataType: "json",
-  });
+    return $.ajax({
+        type: "GET",
+        url: "/building_map/markers",
+        dataType: "json",
+    });
 }
 
 async function getView() {
-  return $.ajax({
-    type: "GET",
-    url: "/building_map/view",
-    dataType: "json",
-  });
+    return $.ajax({
+        type: "GET",
+        url: "/building_map/view",
+        dataType: "json",
+    });
 }
 
 function setupGeoJsonFeature(feature, layer) {
-  const level = parseInt(feature.properties.level_name);
+    const level = parseInt(feature.properties.level_name);
+
+    if (isNaN(level)) {
+        console.warn(
+            "feature has no valid level name and will be skipped: ",
+            feature.properties.level_name,
+            feature
+        );
+        return;
+    }
 
-  if (isNaN(level)) {
-    console.warn(
-      "feature has no valid level name and will be skipped: ",
-      feature.properties.level_name,
-      feature
-    );
-    return;
-  }
-
-  if (!floors[level]) {
-    floors[level] = { rooms: L.layerGroup(), labels: L.layerGroup() };
-    const newLayer = L.layerGroup([floors[level].rooms, floors[level].labels]);
-    layerControl.addBaseLayer(newLayer, feature.properties.level_name);
-    // We add the current floor to the map here so that the map and layer control reference the same object
-    // now the layer control will select the correct check box automatically
-    if (level === currentFloor) {
-      newLayer.addTo(map);
+    if (!floors[level]) {
+        floors[level] = { rooms: L.layerGroup(), labels: L.layerGroup() };
+        const newLayer = L.layerGroup([floors[level].rooms, floors[level].labels]);
+        layerControl.addBaseLayer(newLayer, feature.properties.level_name);
```

Comment:
Could we somehow manage to order these layers? The order currently is `0` -> `1` -> `2` -> `-1`.